libbpf-tools/cgroup_helpers.c

```diff
@@ -0,0 +1,291 @@
+// SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause)
+/* Copyright (c) 2025 Rong Tao */
+#include <assert.h>
+#include <dirent.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+#include <string.h>
+#include "cgroup_helpers.h"
+
+
+int cgroup_get_roots(char ***roots)
+{
+	char line[2048];
+	char fsname[128];
+	char mountpoint[PATH_MAX];
+	char fstype[64];
+	char mntoptions[256];
+	int dump_frequency;
+	int fsck_order;
+	FILE *fp;
+	int n = 0;
+
+	fp = fopen("/proc/mounts", "r");
+	if (!fp)
+		return -errno;
+
+	*roots = NULL;
+
+	while (fgets(line, sizeof(line), fp)) {
+		if (sscanf(line, "%s %s %s %s %d %d\n", fsname, mountpoint,
+			fstype, mntoptions, &dump_frequency,
+			&fsck_order) != 6)
+			continue;
+
+		/* Only need cgroup or cgroup2 */
+		if (strcmp(fstype, "cgroup") && strcmp(fstype, "cgroup2"))
+			continue;
+
+		n++;
+		*roots = (char **)realloc(*roots, n * sizeof(char *));
+		(*roots)[n - 1] = strdup(mountpoint);
+	}
+
+	fclose(fp);
+
+	return n;
+}
+
+int cgroup_free_roots(char **roots, int nentries)
+{
+	int i;
+
+	if (!roots || nentries <= 0)
```

Comment:
Looks like 'nentries <= 0' is not needed. ir roots is not NULL, there should be at least one entry, right?